client/lib/analytics/cart.js

```diff
@@ -10,12 +10,16 @@ import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import { recordAddToCart } from 'calypso/lib/analytics/record-add-to-cart';
 import { getAllCartItems } from 'calypso/lib/cart-values/cart-items';
 
+function difference( items1, items2 ) {
+	return items1.filter( ( a ) => ! items2.some( ( b ) => isEqual( a, b ) ) );
+}
+
 export function recordEvents( previousCart, nextCart ) {
 	const previousItems = getAllCartItems( previousCart );
 	const nextItems = getAllCartItems( nextCart );
 
-	each( differenceWith( nextItems, previousItems, isEqual ), recordAddEvent );
-	each( differenceWith( previousItems, nextItems, isEqual ), recordRemoveEvent );
+	each( difference( nextItems, previousItems, isEqual ), recordAddEvent );
+	each( difference( previousItems, nextItems, isEqual ), recordRemoveEvent );
```

Comment:
we don't need to pass the equality function here anymore, right?

Reply:
I think we still have to pass it, because we're comparing cart items, and those are objects.

Reply:
The new `difference` function has `isEqual` built in and doesn't accept a 3rd argument -- maybe I'm missing something?

Reply:
Ah, so this is what you meant 🤦 . Of course, we can totally ditch that, good catch! Fixing

Reply:
Done in f8e297727092ec3f4fcaf78b3f854c123c3fce21